lua/swenv/api.lua

```diff
@@ -193,11 +194,17 @@ M.auto_venv = function()
 
   local project_dir, _ = project_nvim.get_project_root()
   if project_dir then -- project_nvim.get_project_root might not always return a project path
-    local project_venv_name = read_venv_name(project_dir)
-    if not project_venv_name then
+    local venv_data = get_project_venv_data(project_dir)
+    if not venv_data then
       return
     end
-    local closest_match = best_match(venvs, project_venv_name)
+    if type(venv_data) == type({}) then
+      vim.list_extend(venvs, venv_data)
+      set_venv(venv_data)
```

Comment:
would be nicer if this and the below could be unified, see my comment about the return type of `get_project_venv_data`